atc/gc/pipeline_collector.go

```diff
@@ -0,0 +1,33 @@
+package gc
+
+import (
+	"context"
+
+	"code.cloudfoundry.org/lager/lagerctx"
+	"github.com/concourse/concourse/atc/db"
+)
+
+type pipelineCollector struct {
+	pipelineLifecycle db.PipelineLifecycle
+}
+
+func NewPipelineCollector(pipelineLifecyle db.PipelineLifecycle) *pipelineCollector {
+	return &pipelineCollector{
+		pipelineLifecycle: pipelineLifecyle,
+	}
+}
+
+func (pc *pipelineCollector) Run(ctx context.Context) error {
+	logger := lagerctx.FromContext(ctx).Session("pipeline-collector")
+
+	logger.Debug("start")
+	defer logger.Debug("done")
+
+	_, err := pc.pipelineLifecycle.ArchiveAbandonedPipelines()
```

Comment:
Maybe we should log the number of pipelines archived? Just logging start and done doesn't give much visibility into when things actually get archived

Reply:
We debated this and figured that it's more valuable as a metric instead of a log line. We didn't discuss or dig into trying to add the metric though.

Reply:
I was thinking about emitting metrics vs logging the number, don't really feel strongly about either

Reply:
Good point, metrics makes sense - looks like there's precedence for this with e.g. the `ChecksCollector`:

https://github.com/concourse/concourse/blob/669f6b20a4b677443f70c997ae3eb80815726179/atc/gc/checks_collector.go#L36